docs/file-reference.md

```diff
@@ -0,0 +1,151 @@
+# OpenCompose file reference documentation
+
+
+OpenCompose file example
+```yml
+version: "0.1-dev"
+
+services:
+   name: foobar
+   containers:
+   - image: foo/bar:tag
+     env:
+     - foo=bar
+     command: ["/bin/foobar"]
+     args: 
+     - "-f"
+     - "/tmp/foobar"
+     ports:
+     - port: 8080:80
+
+```
+
+
+OpenCompose file has two main sections version and services.
+
+## version
+OpenCompose format version
+
+## services
+`services` is main section that lists all the services that this OpenCompose files describes.
+`services` has to be array of [ServiceSpec](#servicespec)
+
+
+### ServiceSpec
+This describes one service. One service doesn't necessarily mean one container.
+One service can be comprised of multiple containers.
+Each service has name and list of the containers.
+
+```yml
+# <ServiceSpec>
+  name: foo
+  containers:
+  - <ContainerSpec>
+  - <ContainerSpec>
+```
+
+#### name
+| type | required |
+|------|----------|
+|string|    yes   |
+
+Name of the service.
+
+### containers
+| type                                    | required |
+|-----------------------------------------|----------| 
+|array of [ContainerSpec](#containerspec) |    yes   |
+
+Each item in [containers](#containers) ([ContainerSpec](#containerspec)) defines one container.
+All the containers in container array for given service will be in same Pod.
+
+
+### ContainerSpec
+```yml
+# <ContainerSpec>
+  image: foo/bar:tag
+  env:
+  - VARIABLE=value
+  command:
+    - /foo/bar
+  args:
+    - -foo
+    - -bar
```

Comment:
This double dash thing seems suspicious although it should be a valid syntax. Can we put something nice here like `- "--file"` `- "filename"`